frontend-react/src/hooks/network/ServiceHooks.ts

```diff
@@ -0,0 +1,109 @@
+import {
+    Dispatch,
+    SetStateAction,
+    useCallback,
+    useEffect,
+    useMemo,
+    useState,
+} from "react";
+import { AccessToken } from "@okta/okta-auth-js";
+import _ from "lodash";
+
+import { auxExports } from "../UseCreateFetch";
+import {
+    MembershipState,
+    MemberType,
+    ServiceSettings,
+} from "../UseOktaMemberships";
+import { RSService, servicesEndpoints } from "../../config/endpoints/settings";
+import { RSNetworkError } from "../../utils/RSNetworkError";
+
+type ServiceState = ServiceSettings & {
+    setActiveService: Dispatch<SetStateAction<string>>;
+};
+
+/** Prunes all settings data not relevant to our application session */
+export const cleanServicesArray = (arr: RSService[]) => {
+    return arr.map((obj) =>
+        _.pick(obj, "name", "organizationName", "topic", "customerStatus")
+    );
+};
+
+/** Fetches membership services (senders, receivers) from the ReportStream API. Updates
+ * whenever the given {@link MembershipState} or AccessToken (Okta) are updated.
+ * @remarks ONLY FOR USE WITHIN {@link SessionContext} */
+export const useMemberServices = (
+    state: MembershipState,
+    token: AccessToken | undefined
+): ServiceState => {
+    const [senders, setSenders] = useState<RSService[] | undefined>();
+    const [receivers, setReceivers] = useState<RSService[] | undefined>();
+    const [activeService, setActiveService] = useState<string>("default");
+    /* Because this is used at the SessionContext > useOktaMemberships level, it does not have access
```

Comment:
Hmmm I'm wondering if we need to invoke it directly in useOktaMemberships or if we can use a lazy-fetch approach instead wherein useMemberServices can be used anywhere we need to have access to the list of services -- we could leverage 1) prepopulating the react-query cache through values in LocalStorage if applicable, 2) add the organization `parsedName` as part of the query key, and 3) make any hook that requires the list of services a dependent query or hide anything visual behind a loader.  This'd help decouple this from useOktaMemberships and obviate the need for the custom fetch since any invocations would be within the SessionContext 🤔

Reply:
Given some thought, we _could_ use React Query for this given the provider swap in the PR, but just not with the AuthFetchProvider tooling (i.e. `rsUseQuery`); we'd still have to do the auth leg work. As for lazy fetching, I'm not seeing the benefit -- it'd make the first load of a service-based feature like Daily Data equally as long as it takes now, wouldn't it?

Reply:
After toying around with using `useQuery` for this, it comes with a slew of issues to troubleshoot. I vote we kick that can since this implementation works fine.

Reply:
The idea here would be only loading what we need/making requests to the API on demand rather than at user login and treating services as an association rather than conflating them with session logic.  As far as I can tell, we don't need services to be loaded "as part of" the user as soon as they're logged in since they're not directly related to authentication (if we did, we should update the initial login payload to include them).  I don't think these two things need to be tightly coupled together 🤔

Reply:
My thought was that showing/hiding navigation UI based on available services was a more generalized approach than segregating experience by `memberType` as we do now; it'd remove the need to infer from an organization name in Okta what features should appear, and mean that any user with an account is a `MemberType.STANDARD` unless they're a `PRIME_ADMIN`. I've never been a fan of our need to parse a name prefix since it creates another nomenclature standard that we need to maintain.

So, in practice, a user is in a single `DHmy-org` group in Okta, and the show/hide of any features depends on whether they are given one or more settings for a type of service. To an extent, including pipeline functionality, _all_ features do require a service, either sender or receiver.

Reply:
Right, but this still seems like we're conflating two separate concepts here: authentication versus associations.  This is how I understand how things are currently laid out (feel free to correct me if I'm wrong here!):

- An end user _is an_ Organization
- An Organization _has many_ services
- A service _has many_ reports, files, et cetera

And there are two types of settings: one at an Organization level and one at a service level.  (A similar analogy I can think of is having different profiles on a social media platform: the user logs in with one set of credentials but can swap between different profiles, each with their own memberships, followers, et cetera.)  I think we should keep these logically separate because conflating them means that we're going to be treating an Organization like a service and vice versa when they really operate on two different levels.  Toggling features based on services isn't a need I'm aware of, and given the recent discussion and implementation behind feature flags, it seems like we'd only want to do this on an Organization level anyway.

Additionally, bypassing react-query here means that we're not going to be able to take advantage of the query cache.  We're going to be making the potentially time-consuming requests to fetch the senders and receivers _every_ time we impersonate a new Organization, even if we had already impersonated that Organization beforehand:
- Log in as admin --> no sender/receiver fetches
- Impersonate `abbott` --> immediate, uncached sender/receiver fetches
- Impersonate `ignore` --> immediate, uncached sender/receiver fetches
- Impersonate `abbott` again --> immediate, uncached sender/receiver fetches
Making these requests lazy react-query queries means that 1) we won't be making immediate requests until the user needs them and 2) we won't be making redundant requests since that data will already exist on the client side.

Reply:
Also, I agree with you about wanting to not base user ability on the prefix of the group, but we already have working functional precedent on it -- in both the front-end code and in the back-end code (for [scopes](https://github.com/CDCgov/prime-reportstream/blob/master/prime-router/src/main/kotlin/tokens/Scope.kt#L176-L204) and [authentication](https://github.com/CDCgov/prime-reportstream/blob/master/prime-router/src/main/kotlin/tokens/OktaAuthentication.kt#L92-L135)).  If we want to deviate from this pattern, I'd argue that it should be a joint movement on both parts of the stack or we'll be introducing unexpected behavior.